### PR TITLE
Calculating area of circle

### PR DESCRIPTION
**Pull request by raghad598**
---
name: Calculating area of circle
about: calculating area of circle is a function that computes the area of a circle given its radius
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called at the top level of the function file 
  - _Recursive solutions **can** call the function from **inside** the function body_

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by HebaShaheen** [comment]
Good job Raghad. Here are some changes you have to do.
- change the function name "calculate_circle_area" to be same as the file name "area_of_circle.py"
- for this doctest you could replace it with another test that pass. don't put defensive tests here
>>> calculate_circle_area(-1)
    Traceback (most recent call last):
        ...
    ValueError: Radius cannot be negative
-for the test file put docstring in the class and for each unit test

**Comment by raghad598** [comment]
got it heba i'll do the changes

**Review by HebaShaheen — DISMISSED** [review]

**Review by HebaShaheen — APPROVED** [review]

**Review by HebaShaheen — APPROVED** [review]

**Review by HebaShaheen — APPROVED** [review]

File: solutions/area_of_circle.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    >>> area_of_circle(0)
+    0.0
+    >>> area_of_circle(3.5)
+    38.48451000647496
+    """
+    if radius < 0:
+        raise ValueError("Radius cannot be negative")
+
+    return math.pi * radius**2
+
```
Comment (Moealfadil) [inline solutions/area_of_circle.py:36]: its good to add comments to show the strategy

File: solutions/tests/test_area_of_circle.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        self.assertAlmostEqual(area_of_circle(10), 314.1592653589793)
+
+    def test_zero_radius(self):
+        """Test the function with a zero radius."""
+        self.assertEqual(area_of_circle(0), 0.0)
+
+    def test_large_radius(self):
+        """Test the function with a very large radius."""
+        self.assertAlmostEqual(area_of_circle(1000), 3141592.653589793)
+
```
Comment (Moealfadil) [inline solutions/tests/test_area_of_circle.py:20]: its a good idea to add some  tests for defensive assertions

**Review by Moealfadil — APPROVED** [review]
looks all good. well done!
